datastream-server/src/main/java/com/linkedin/datastream/server/assignment/LoadBasedPartitionAssigner.java

```diff
@@ -80,40 +80,36 @@ public LoadBasedPartitionAssigner(int defaultPartitionBytesInKBRate, int default
       ClusterThroughputInfo throughputInfo, Map<String, Set<DatastreamTask>> currentAssignment,
       List<String> unassignedPartitions, DatastreamGroupPartitionsMetadata partitionMetadata, int maxPartitionsPerTask) {
     String datastreamGroupName = partitionMetadata.getDatastreamGroup().getName();
-    Map<String, PartitionThroughputInfo> partitionInfoMap = throughputInfo.getPartitionInfoMap();
+    Map<String, PartitionThroughputInfo> partitionInfoMap = new HashMap<>(throughputInfo.getPartitionInfoMap());
     Set<String> tasksWithChangedPartition = new HashSet<>();
 
     // filter out all the tasks for the current datastream group, and retain assignments in a map
-    Map<String, Set<String>> newPartitions = new HashMap<>();
+    Map<String, Set<String>> newPartitionAssignmentMap = new HashMap<>();
     currentAssignment.values().forEach(tasks ->
         tasks.forEach(task -> {
           if (task.getTaskPrefix().equals(datastreamGroupName)) {
             Set<String> retainedPartitions = new HashSet<>(task.getPartitionsV2());
             retainedPartitions.retainAll(partitionMetadata.getPartitions());
-            newPartitions.put(task.getId(), retainedPartitions);
+            newPartitionAssignmentMap.put(task.getId(), retainedPartitions);
             if (retainedPartitions.size() != task.getPartitionsV2().size()) {
               tasksWithChangedPartition.add(task.getId());
             }
           }
     }));
 
     int numPartitions = partitionMetadata.getPartitions().size();
-    int numTasks = newPartitions.size();
+    int numTasks = newPartitionAssignmentMap.size();
     validatePartitionCountAndThrow(datastreamGroupName, numTasks, numPartitions, maxPartitionsPerTask);
 
     // sort the current assignment's tasks on total throughput
     Map<String, Integer> taskThroughputMap = new HashMap<>();
     PartitionThroughputInfo defaultPartitionInfo = new PartitionThroughputInfo(_defaultPartitionBytesInKBRate,
         _defaultPartitionMsgsInRate, "");
 
-    newPartitions.forEach((task, partitions) -> {
+    newPartitionAssignmentMap.forEach((task, partitions) -> {
       int totalThroughput = partitions.stream()
           .mapToInt(p ->  {
-            int index = p.lastIndexOf('-');
-            String topic = p;
-            if (index > -1) {
-              topic = p.substring(0, index);
-            }
+            String topic = extractTopicFromPartition(p);
```

Comment:
Probably not a issue but for my understanding.
Below (for unassigned partitions for loop) we check if a partition is present in partitinInfoMap and if not check if partition's topic is present in partitionInfoMap. Why dont we do it that way here?

Reply:
We are doing it, see the default value. This loop is for the partitions already assigned. That's why it is of no value to update the map.